id: PYSEC-2020-240
details: A Denial of Service (DoS) condition is possible in OpenStack Mistral in versions
  up to and including 7.0.3. Submitting a specially crafted workflow definition YAML
  file containing nested anchors can lead to resource exhaustion culminating in a
  denial of service.
affected:
- package:
    name: mistral
    ecosystem: PyPI
    purl: pkg:pypi:mistral
  ranges:
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 7.0.4
  versions:
  - "0"
  - 0.0.1
  - 0.0.2
  - 0.0.3
  - 0.0.4
  - "0.1"
  - 0.1.1
  - 1.0.0
  - 1.0.0.0b1
  - 1.0.0.0b2
  - 1.0.0.0b3
  - 1.0.0.0rc1
  - 1.0.0.0rc2
  - 1.0.1
  - 2.0.0
  - 2.0.0.0b1
  - 2.0.0.0b2
  - 2.0.0.0b3
  - 2.0.0.0rc1
  - 2.0.0.0rc2
  - 2.0.0.0rc3
  - 3.0.0.0b1
  - 3.0.2
  - 4.0.0
  - 4.0.0.0b1
  - 4.0.0.0b2
  - 4.0.0.0b3
  - 4.0.0.0rc1
  - 4.0.0.0rc2
  - 4.0.1
  - 4.0.2
  - 5.0.0
  - 5.0.0.0b1
  - 5.0.0.0b2
  - 5.0.0.0b3
  - 5.0.0.0rc1
  - 5.1.0
  - 5.2.0
  - 5.2.1
  - 5.2.2
  - 5.2.3
  - 5.2.4
  - 5.2.5
  - 5.2.6
  - 5.2.7
  - 5.2.8
  - 6.0.0
  - 6.0.0.0b1
  - 6.0.0.0b2
  - 6.0.0.0b3
  - 6.0.0.0rc1
  - 6.0.1
  - 6.0.2
  - 6.0.3
  - 6.0.4
  - 6.0.5
  - 6.0.6
  - 7.0.0
  - 7.0.0.0b1
  - 7.0.0.0b2
  - 7.0.0.0b3
  - 7.0.0.0rc1
  - 7.0.1
  - 7.0.2
  - 7.0.3
references:
- type: REPORT
  url: https://bugzilla.redhat.com/show_bug.cgi?id=1645332
- type: WEB
  url: https://bugs.launchpad.net/mistral/+bug/1785657
aliases:
- CVE-2018-16848
modified: "2021-08-27T03:22:07.520532Z"
published: "2020-06-15T15:15:00Z"
